atc/db/pipeline_pauser.go

```diff
@@ -0,0 +1,92 @@
+package db
+
+import (
+	"strconv"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/concourse/concourse/atc/db/lock"
+)
+
+//counterfeiter:generate . PipelinePauser
+type PipelinePauser interface {
+	PausePipelines(daysSinceLastBuild int) error
+}
+
+func NewPipelinePauser(conn Conn, lockFactory lock.LockFactory) PipelinePauser {
+	return &pipelinePauser{
+		conn:        conn,
+		lockFactory: lockFactory,
+	}
+}
+
+type pipelinePauser struct {
+	conn        Conn
+	lockFactory lock.LockFactory
+}
+
+func (p *pipelinePauser) PausePipelines(daysSinceLastBuild int) error {
+	activePipelines, err := getActivePipelines(p.conn, daysSinceLastBuild)
+	if err != nil {
+		return err
+	}
+
+	rows, err := pipelinesQuery.Where(sq.And{
+		sq.Eq{
+			"p.paused": false,
+		},
+		sq.NotEq{
```

Comment:
Hmm does something like Where(sq.Expr("SELECT p.id FROM pipelines AS p LEFT JOIN jobs AS j ON j.pipeline_id = p.id WHERE j.last_scheduled > CURRENT_DATE - ?::INTERVAL", strconv.Itoa(days) + " day)). not work? This would help us not need to pass the pipelines over from one query to another, which can sometimes be slow if there are a lot active pipelines.

Reply:
The query you have there is what I started with but squirrel would not interpolate the `'? day'` part EVER. I rewrote that query SO MANY TIMES (nearly lost my sanity) but sadly ended up with the two-part query that got merged.

The only thing that makes me somewhat okay with this implementation is the fact that this query is only run once every 24hrs.

That said, I'll give your suggestion a try because I would also REALLY like to get rid of the second query and have it as a subquery instead.

Reply:
yay it worked! PR incoming 😄

Reply:
Yaay!! HMM I wonder what made it work this time with the `? day` 🤔

Reply:
I think the problem was with the way I was writing the query with squirrel vs when I was manually figuring out the query in postico. What I kept trying to do was get squirrel/postgres to interpolate the value for me (you know, avoid sql injection). So I kept writing the query as `... '? day'::INTERVAL`.

What your suggestion was changed the query from `... '? day'::INTERVAL` to `...?::INTERVAL` and TADA everything worked. I was too hung up on trying to pass ONLY the int instead of a string that contained the int.

Reply:
Ohh I see! Cool thats good to know haha! Thanks for explaining :)